// Copyright (C) 2016-2019,2021 Rodrigo Jose Hernandez Cordoba
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

syntax = "proto3";
package AeonGames;

import "vector2.proto";
import "vector3.proto";
import "vector4.proto";
import "matrix4x4.proto";

// TODO Consider renaming to UniformVariableMsg or similar, since that's what this really is
message PropertyMsg {
	string name = 1;
	oneof value
	{
		float           scalar_float = 2;
		uint32          scalar_uint  = 3;
		sint32          scalar_int   = 4;
		Vector2Msg   vector2      = 5;
		Vector3Msg   vector3      = 6;
		Vector4Msg   vector4      = 7;
		Matrix4x4Msg matrix4x4    = 8;
	}
}
